src/components/TodoFooter/TodoFooter.tsx

```diff
@@ -0,0 +1,91 @@
+import { useContext, useMemo } from 'react';
+import classnames from 'classnames';
+import { Status } from '../../types/Status';
+import { FilterContext } from '../../context/FilterContext';
+import { TodoContext } from '../../context/TodoContext';
+import { deleteTodo } from '../../api/todos';
+import { ErrorContext } from '../../context/ErrorContext';
+
+export const TodoFooter = () => {
+  const { todos, setTodos } = useContext(TodoContext);
+  const { selectedFilter, setSelectedFilter } = useContext(FilterContext);
+  const { setErrorMessage } = useContext(ErrorContext);
+
+  const hasCompletedTodo = useMemo(() => {
+    return todos.some(todo => todo.completed);
+  }, [todos]);
+
+  const clearCompleted = () => {
+    todos.forEach(({ id, completed }) => {
+      if (completed) {
+        deleteTodo(id)
+          .then(() => {
+            setTodos(prevState => prevState.filter(todo => todo.id !== id));
+          })
+          .catch(() => setErrorMessage('Unable to delete a todo'));
+      }
+    });
```

Comment:
Don't use `forEach` for async operations, use `for of` with `async/await` or `collect requests with `map` and then `Promise.all`.

Also, add loaders to notify users what todos will be deleted.